structs/map_test.go

```diff
@@ -0,0 +1,39 @@
+package structs
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+type structTest struct {
+	Test string `field:"test"`
+}
+
+func TestMap(t *testing.T) {
```

Comment:
Please add tests for partial conversions where not all fields are converted.

Reply:
Care to give me an example of what partial conversion is?

Reply:
Certainly

```go
type TestStruct struct {
	Field1 int `map:"field1"`
	Field2 string `map:"field2"`
	Field3 bool
	Field4 TestStruct
}
```

`struct.Map(ts, "map")` shoudl result in

```go
map[string]any{
	"field1": <value>,
	"field2": <value>,
}
```